### ISSUE TITLE
Add support for managing kubelet service

### ISSUE THREAD

**Issue by CharlieTLe** [open-telemetry/opentelemetry-operator#3277]
### Component(s)

collector, target allocator

### Is your feature request related to a problem? Please describe.

I would like to use a ServiceMonitor to scrape metrics from all of the kubelets running in a cluster. The kubelet is usually exposed on each node on port 10250. The ServiceMonitor and PodMonitor only support selecting services or pods, but the metrics from the kubelet isn't exposed via a pod.

### Describe the solution you'd like

The prometheus-operator can manage a selector-less service that updates the endpoints that map to the kubelet for each node. This service can be used by a ServiceMonitor and then by the target allocator to distribute the targets across collectors to scrape.

This same logic can be imported into the opentelemetry-operator to manage the kubelet service.

### Describe alternatives you've considered

The alternative to this appears to be creating a daemonset of collector pods that runs on each node to collect the kubelet metrics, but for clusters with many nodes, running a small collector that just scrapes the kubelet seems inefficient.

### Additional context

Here's how the prometheus-operator does it: https://github.com/prometheus-operator/prometheus-operator/blob/1dc7f9215a64d6611e7ac7d278c7c2cc41b95c44/pkg/kubelet/controller.go#L273-L301

**Comment by jaronoff97** [comment]
ha i've been avoiding this one because I don't love the idea of the logic living here too... I got around this by taking the rendered scrape_configs from the servicemonitor and creating an explicit [scrape config](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-kube-stack/daemon_scrape_configs.yaml#L72-L176) from it. If we were to support the ScrapeConfig CRD (this issue: #3103) would that combination be sufficient to solve the underlying issue here. 

You mention not wanting to run a DS which is fine, you would just need to update that scrape config to use the node role ([described here](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#node)) for the kubernetes_sd_config.

**Comment by CharlieTLe** [comment]
> If we were to support the ScrapeConfig CRD (this issue: https://github.com/open-telemetry/opentelemetry-operator/pull/3103) would that combination be sufficient to solve the underlying issue here?

Yeah, I think using the ScrapeConfig CRD and the node role from kubernetes_sd_config would be better than managing a service for scraping the kubelet on each of the nodes. This way, we could make use of the target allocator to distribute the targets to the collectors.

**Comment by CharlieTLe** [comment]
The ScrapeConfig CR should be able to do this now.
